test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -468,6 +468,11 @@ public static void setContentType() throws Exception {
         Requests.INDEX_CONTENT_TYPE = randomFrom(XContentType.values());
     }
 
+    @AfterClass
+    public static void resetRootLogger(){
+        Loggers.setLevel(LogManager.getLogger(), Level.INFO);
```

Comment:
Should we first capture what the original level was at and then reset to _that_? I'm sure there are scenarios where folks run tests with `-Dtests.logger.level` and we don't want to overwrite that.

Reply:
An alternative might be this approach https://github.com/elastic/elasticsearch/pull/106260

Reply:
I think we still want your PR @mosche

Reply:
The intention here is for it to be a catch-all.

Reply:
What I meant is, we could get the state of all loggers in @BeforeClass and revert to that state in @AfterClass as done in https://github.com/elastic/elasticsearch/pull/106260.
`Loggers.setLevel` overrides all child levels as well, there's a bunch of loggers configured at DEBUG if I recall right which would be impacted by this as well.

Reply:
good point. I would expect that addivity being false should prevent this. We set this in our log4j2 config file, but not sure it is used in tests.
I will follow up on this

Reply:
But I suppose all of that should be done based on annotations on the test level, so this should be fine 👍

Reply:
>  I would expect that addivity being false should prevent this.

We're deliberately doing that ourselves in `Loggers`:
https://github.com/elastic/elasticsearch/blob/405b88b882f6279fbd46dcec4413e403eec77a8c/server/src/main/java/org/elasticsearch/common/logging/Loggers.java#L184-L190

Reply:
Another point to verify @pgomulka , `@TestLogging` is also allowed on the class level, e.g.
```
@TestLogging(reason = "these tests do a lot of log-worthy things but we usually don't care", value = "org.elasticsearch:FATAL")
public class AtomicRegisterCoordinatorTests extends CoordinatorTests
```

We should have a look at how that configuration is applied and make sure it doesn't conflict with this. E.g. assume one sets the root level using `@TestLogging`, is this reading the root level before or after the modification?

Reply:
Perhaps we should just having `LoggingListener` do this so these things don't step on eachother. Seems logical to place this kind of logic there, rather than in setup/teardown methods in `ESTestCase` where ordering isn't guaranteed.

Reply:
+1 to doing this in LoggingListener, it should be more robust.